standards/SPIFFE_Federation.md

```diff
@@ -0,0 +1,316 @@
+
+As an interoperability concern, servers MUST NOT require client authentication to access the bundle endpoint; this includes both transport layer (e.g. client certificates) and HTTP-layer (e.g. Authentication headers) authentication schemes.
+
+Upon receiving an HTTP GET request for the correct path, the bundle endpoint server MUST respond with the most up-to-date version of the SPIFFE bundle available. The exact path value may be chosen by the operator, and appears as part of the bundle endpoint URL.
+
+Bundle endpoint servers MAY respond with HTTP redirect (as defined by [RFC 7231 section 6.4][8]) if the authority for serving the requested bundle has moved.  The target URL of the redirect MUST also be a [valid bundle endpoint URL](#5211-endpoint-url-requirements) as defined in this profile. Servers SHOULD use a temporary redirect; support for redirection is intended for operational considerations (e.g. serving a bundle via a CDN) and not as a means to permanently migrate the bundle endpoint URL. See the [Security Considerations](#7-security-considerations) for more information.
+
+##### 5.2.1.4. Consuming a Bundle Endpoint
+SPIFFE bundle endpoint clients utilize standard TLS-protected HTTP (i.e. HTTPS) when interacting with an `https_web` bundle endpoint. When connecting to the endpoint, the server certificate MUST be validated in accordance with [RFC 6125][11]. To summarize that document, the server certificate MUST be issued by a certificate authority which is locally trusted and it MUST contain an X.509 Subject Alternative Name (or Common Name) that matches the host component of the configured endpoint URL.
+
+After establishing a TLS connection to the bundle endpoint and authenticating the presented server certificate, clients issue an HTTP GET for the path specified by the endpoint URL. The body of the response is a SPIFFE bundle. Clients MUST know the name of the trust domain that the endpoint URL represents prior to retrieving the trust bundle, ideally via explicit configuration; please see the [Security Considerations](#7-security-considerations) section for more information.
+
+Bundle endpoint servers MAY respond with HTTP redirect (as defined by [RFC 7231 section 6.4][8]).  Bundle endpoint clients SHOULD follow the redirect if the URL meets all the requirements of a valid bundle endpoint URL.  When connecting to the new URL, the same TLS considerations MUST be applied as connecting to the original URL.  Bundle endpoint clients SHOULD use the configured endpoint URL for each bundle refresh and SHOULD NOT permanently store the location for future fetches. See the [Security Considerations](#7-security-considerations) for more information. 
+
+#### 5.2.2. SPIFFE Authentication (`https_spiffe`)
```

Comment:
Do we gain anything by requiring that the bundle endpoint's cert be itself an X509 SVID? You could imagine that a client gets a root cert for the bundle endpoint in an out-of-band way and adds it to their root store. Or would that fall under the `https_web` profile?

Reply:
I'm assuming here that in your scenario the client checks the domain name from the URI against the Common Name or DNS Subject Alt Name on the cert. In which case, yes, this would be `https_web`.

In `https_spiffe` the client checks the configured SPIFFE ID against the URI Subject Alt Name.